music21/metadata/__init__.py

```diff
@@ -96,161 +184,499 @@ class Metadata(base.Music21Object):
+    to 'dateCreated').  The old music21 v7 name in these cases is still supported
+    when you use attribute-style access.
+
     >>> md = metadata.Metadata(title='Concerto in F')
     >>> md.title
     'Concerto in F'
 
-    Or by three-letter abbreviation:
+    Or by three-letter abbreviation (these are grandfathered in from music21 v7;
+    abbreviations have not been added for new-in-v8 metadata items):
 
     >>> md = metadata.Metadata(otl='Concerto in F')
     >>> md.otl
     'Concerto in F'
     >>> md.title
     'Concerto in F'
 
-    Or by setWorkId
-
-    >>> md.setWorkId('title', 'Rhapsody in Blue')
-    >>> md.title
-    'Rhapsody in Blue'
+    Or set an iterable of values or get a tuple full of (richer-typed) values using
+    dictionary-style access.
 
-    >>> md.otl
-    'Rhapsody in Blue'
+    >>> md = metadata.Metadata()
+    >>> md['composer'] = ['Billy Strayhorn', 'Duke Ellington']
+    >>> md['composer']
+    (<music21.metadata.primitives.Contributor composer:Billy Strayhorn>,
+    <music21.metadata.primitives.Contributor composer:Duke Ellington>)
+    >>> md.composer
+    'Billy Strayhorn and Duke Ellington'
+    >>> md.contributors
+    [<music21.metadata.primitives.Contributor composer:Billy Strayhorn>,
+    <music21.metadata.primitives.Contributor composer:Duke Ellington>]
 
-    >>> md.composer = 'Gershwin, George'
+    searchAttributes are used by .search() methods to determine what attributes can
+    be searched for.  There are some synonyms (like 'dateCreated' and 'date')
+    that will find the same items.  This is because some uniqueNames have been
+    updated in music21 v8 ('dateCreated'), and the old v7 name ('date') has
+    been grandfathered in as a synonym.
 
-    These are used by .search() methods to determine what attributes are
-    made available by default.
+    allUniqueNames is a list of all the unique names (without any of the
+    grandfathered-in v7 workId synonyms).
 
     >>> md.searchAttributes
-    ('actNumber', 'alternativeTitle', 'associatedWork', 'collectionDesignation',
-     'commission', 'composer', 'copyright', 'countryOfComposition', 'date', 'dedication',
-     'fileFormat', 'fileNumber', 'filePath',
-     'groupTitle', 'localeOfComposition', 'movementName', 'movementNumber', 'number',
-     'opusNumber', 'parentTitle', 'popularTitle', 'sceneNumber', 'textLanguage',
-     'textOriginalLanguage', 'title', 'volume')
-
-    Plus anything that is in contributors...
-
-    All contributors are stored in a .contributors list:
-
-    >>> md.contributors
-    [<music21.metadata.primitives.Contributor composer:Gershwin, George>]
+    ('abstract', 'accessRights', 'accompanyingMaterialWriter', 'accrualMethod',
+    'accrualPeriodicity', 'accrualPolicy', 'actNumber', 'actor', 'adapter',
+    'afterwordAuthor', 'alternativeTitle', 'animator', 'annotator', 'architect',
+    'arranger', 'artist', 'associatedWork', 'attributedComposer', 'audience',
```

Comment:
this list should be trimmed down both internally and in the docs. -- it's intimidating to see such a long list when a user is trying to grasp what this module does.

Reply:
And it'll be a pain if it updates later.

Reply:
ok

Reply:
done.